File: src/main/java/ru/arc/xserver/announcements/AnnouncementMessager.java
package ru.arc.xserver.announcements;

import ru.arc.ARC;
import ru.arc.network.ChannelListener;
import ru.arc.util.Common;
import ru.arc.xserver.XMessage;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.concurrent.CompletableFuture;

@RequiredArgsConstructor
public class AnnouncementMessager implements ChannelListener {

    @Getter
    final String channel;

    @Override
    public void consume(String channel, String message, String server) {
        XMessage data = Common.gson.fromJson(message, XMessage.class);
        AnnounceManager.queue(data);
    }


    public void send(XMessage data) {
        CompletableFuture.supplyAsync(() -> Common.gson.toJson(data))
                .thenAccept(json -> ARC.redisManager.publish(channel, json));
    }
}

File: src/main/kotlin/ru/arc/xserver/announcements/AnnouncementMessager.kt
package ru.arc.xserver.announcements

import ru.arc.ARC
import ru.arc.network.ChannelListener
import ru.arc.util.Common
import ru.arc.xserver.XMessage
import java.util.concurrent.CompletableFuture

class AnnouncementMessager(val channel: String) : ChannelListener {

    override fun consume(channel: String, message: String, server: String) {
        val data = Common.gson.fromJson(message, XMessage::class.java)
        AnnounceManager.queue(data)
    }

    fun send(data: XMessage) {
        CompletableFuture.supplyAsync { Common.gson.toJson(data) }
            .thenAccept { json -> ARC.redisManager!!.publish(channel, json) }
    }
}
